core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -0,0 +1,150 @@
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for lesson information card modal.
+ */
+
+import { Component } from '@angular/core';
+import { NgbActiveModal, NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import { StateCard } from 'domain/state_card/state-card.model';
+import { StoryPlaythrough } from 'domain/story_viewer/story-playthrough.model';
+import { StoryViewerBackendApiService } from 'domain/story_viewer/story-viewer-backend-api.service';
+import { LearnerExplorationSummaryBackendDict } from
+  'domain/summary/learner-exploration-summary.model';
+import { UrlService } from 'services/contextual/url.service';
+import { I18nLanguageCodeService, TranslationKeyType } from
+  'services/i18n-language-code.service';
+
+ @Component({
+   selector: 'oppia-lesson-information-card-modal',
+   templateUrl: './lesson-information-card-modal.component.html'
+ })
+export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
+  // These properties below are initialized using Angular lifecycle hooks
+  // where we need to do non-null assertion. For more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  storyTitleTranslationKey!: string;
+  storyPlaythroughObject!: StoryPlaythrough;
+  storyId!: string;
+  storyTitle: string = '';
+  topicUrlFragment: string;
+  classroomUrlFragment: string;
+  storyUrlFragment: string;
+  expTitleTranslationKey!: string;
+  expDescTranslationKey!: string;
+  displayedCard!: StateCard;
+  explorationId!: string;
+  expTitle!: string;
+  expDesc!: string;
+  contributorNames!: string[];
+  hasStoryTitle!: boolean;
+  chapterTitle!: string;
+  chapterDesc!: string;
+  chapterNumber!: string;
+  numberofCheckpoints!: number;
+  expInfo: LearnerExplorationSummaryBackendDict;
+  startedWidth: number = 0;
+  completedWidth!: number;
+  separatorWidth: number = 0;
+  separatorArray: number[] = [];
+
+
+  constructor(
+    private ngbModal: NgbModal,
+    private ngbActiveModal: NgbActiveModal,
+    private urlService: UrlService,
+    private i18nLanguageCodeService: I18nLanguageCodeService,
+    private storyViewerBackendApiService: StoryViewerBackendApiService,
+  ) {
+    super(ngbActiveModal);
+  }
+
+  ngOnInit(): void {
+    this.explorationId = this.expInfo.id;
+    this.expTitle = this.expInfo.title;
+    this.expDesc = this.expInfo.objective;
+    this.storyId = this.urlService.getUrlParams().story_id;
+    this.hasStoryTitle = (this.storyId !== undefined) ? true : false;
```

Comment:
```suggestion
    this.storyTileIsPresent = !!this.storyId;
```

"hasXXX" sounds like a function name.

Reply:
Done!